### PR TITLE
:running: Update controller runtime, Go 1.12, Cluster API v0.2.5

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

    - Update controller-runtime to v0.3.0
    - Update golang used to v1.12.10
    - Update cluster-api to v0.2.5

### CODE REVIEW

**Comment by chuckha** [comment]
@detiber do you think this needs to go in as a final release to CABPK? i'm inclined to hold off until this repo merges with CAPI

**Comment by detiber** [comment]
> @detiber do you think this needs to go in as a final release to CABPK? i'm inclined to hold off until this repo merges with CAPI

Up to you, I mainly created this because I had already done it for CAPI and CAPA.

**Comment by chuckha** [comment]
I'm unfamiliar with the changes coming in. As far as i know this isn't fixing any bugs we are experiencing. Because of that I will hold off on this PR but gladly accept after we get the merge into CAPI done

**Comment by ncdc** [comment]
+1 to deferring until this repo has moved in to cluster-api if that's ok?

File: go.mod [hunk 1]
```diff
@@ -26,14 +25,17 @@ require (
...
-	sigs.k8s.io/cluster-api v0.2.3
-	sigs.k8s.io/controller-runtime v0.2.2
+	sigs.k8s.io/cluster-api v0.2.5
+	sigs.k8s.io/controller-runtime v0.3.0
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.2.3
+replace (
+	k8s.io/client-go => k8s.io/client-go v0.0.0-20190918200256-06eb1244587a
+	sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.2.5
```
Comment (ncdc) [inline go.mod]: Do we need this?

File: go.mod [hunk 2]
```diff
@@ -26,14 +25,17 @@ require (
...
 	k8s.io/kube-openapi v0.0.0-20190816220812-743ec37842bf // indirect
-	sigs.k8s.io/cluster-api v0.2.3
-	sigs.k8s.io/controller-runtime v0.2.2
+	sigs.k8s.io/cluster-api v0.2.5
+	sigs.k8s.io/controller-runtime v0.3.0
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.2.3
+replace (
+	k8s.io/client-go => k8s.io/client-go v0.0.0-20190918200256-06eb1244587a
```
Comment (ncdc) [inline go.mod]: If we update the client-go version above, do we still need a replace?

File: go.mod (hunk 1 above)
Comment (detiber) [reply go.mod]: I'll try removing this...

File: go.mod (hunk 2 above)
Comment (detiber) [reply go.mod]: Everytime I've tried to remove this previously it reverts back to a different version and introduces build failures.

Comment (ncdc) [reply go.mod]: K. If the version ends up not being `v0.0.0-20190918200256-06eb1244587a`, definitely keep it. But we should update the version up above too, to match, right?

Comment (detiber) [reply go.mod]: Actually, it looks like it's working now, removed both.

**Comment by ncdc** [comment]
LGTM. @vincepri ?

**Pull request merged**